vtkext/private/module/vtkF3DOverlayRenderPass.cxx

```diff
@@ -0,0 +1,229 @@
+#include "vtkF3DOverlayRenderPass.h"
+
+#include <vtkCameraPass.h>
+#include <vtkDefaultPass.h>
+#include <vtkObjectFactory.h>
+#include <vtkOpenGLError.h>
+#include <vtkOpenGLFramebufferObject.h>
+#include <vtkOpenGLQuadHelper.h>
+#include <vtkOpenGLRenderUtilities.h>
+#include <vtkOpenGLRenderWindow.h>
+#include <vtkOpenGLShaderCache.h>
+#include <vtkOpenGLState.h>
+#include <vtkProp.h>
+#include <vtkRenderState.h>
+#include <vtkRenderer.h>
+#include <vtkShaderProgram.h>
+#include <vtkSkybox.h>
+#include <vtkTextureObject.h>
+
+vtkStandardNewMacro(vtkF3DOverlayRenderPass);
+
+void vtkF3DOverlayRenderPass::Render(const vtkRenderState* s)
+{
+  this->Initialize(s);
+
+  double bgColor[3];
+
+  vtkRenderer* r = s->GetRenderer();
+  r->GetBackground(bgColor);
+
+  // force background to full black when generating offscreen layers to avoid blending
+  // problems when compositing layers in the Blend() function
+  r->SetBackground(0.0, 0.0, 0.0);
+
+  vtkRenderState overlayState(s->GetRenderer());
+  overlayState.SetPropArrayAndCount(
+    this->OverlayProps.data(), static_cast<int>(this->OverlayProps.size()));
+  overlayState.SetFrameBuffer(s->GetFrameBuffer());
+
+  this->OverlayPass->Render(&overlayState);
+  r->SetBackground(bgColor);
+
+  this->CompositeOverlay(s);
+}
+
+void vtkF3DOverlayRenderPass::ReleaseGraphicsResources(vtkWindow* w)
+{
+  this->Superclass::ReleaseGraphicsResources(w);
+
+  if (this->QuadHelper)
+  {
+    this->QuadHelper->ReleaseGraphicsResources(w);
+  }
+  if (this->OverlayPass)
+  {
+    this->OverlayPass->ReleaseGraphicsResources(w);
+  }
+  if (this->FrameBufferObject)
+  {
+    this->FrameBufferObject->ReleaseGraphicsResources(w);
+  }
+  if (this->ColorTexture)
+  {
+    this->ColorTexture->ReleaseGraphicsResources(w);
+  }
+}
+
+void vtkF3DOverlayRenderPass::Initialize(const vtkRenderState* s)
+{
+  this->OverlayProps.clear();
+
+  // assign props to the overlay pass
+  vtkProp** props = s->GetPropArray();
+  for (int i = 0; i < s->GetPropArrayCount(); i++)
+  {
+    vtkProp* prop = props[i];
+    if (!vtkSkybox::SafeDownCast(prop) && !vtkProp3D::SafeDownCast(prop))
```

Comment:
I know this code was there already, but `vtkSkybox` derives from `vtkProp3D`, so the first check can be removed.